sympy/algebras/quaternion.py

```diff
@@ -490,6 +490,36 @@ def _ln(self):
 
         return Quaternion(a, b, c, d)
 
+    def evalf(self, *args):
```

Comment:
The `evalf` method should not be overridden as it is provided by the `EvalfMixin` superclass. Instead the `_eval_evalf` method should be defined:
https://github.com/sympy/sympy/blob/5824415f287a1842e47b75241ca4929efd0fbc7b/sympy/sets/sets.py#L652-L653

Reply:
@oscarbenjamin made the changes accordingly and also should any tests be added for this evalf in test_quaternion.py?

Reply:
Yes

Reply:
@oscarbenjamin i've added the tests in the test case function test_quaternion_evalf() but is it correct to create a new test case function or should i add the test cases in test_quaternion_functions()?